### PR TITLE
This PR fixes #4

### LINKED ISSUE #4: Hacktoberfest 2020: Make it easier to install
# Hacktoberfest:  Add install target in Makefile
Currently, there's no way to install `gocat` globally into the system. As such, we need to provide users ability to easily install our program. For example:
```
# This should copy the binary one of the PATH
$ make install
# So user can run the binary as such from anywhere
$ gocat --version
```

#### References: 
- https://github.com/tuladhar/set-brightness/blob/main/Makefile

Puru,
Happy Hacking

### PR DESCRIPTION
**Pull request by vimarsh6739**
This fixes #4 by installing the built binary at /usr/local/bin.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -11,4 +11,4 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
-      - run: make build
```
Comment (tuladhar) [inline .github/workflows/ci.yaml]: We can leave this as it is.

File: Makefile [hunk 2]
```diff
@@ -1,10 +1,16 @@
 #
 # Makefile - `gocat` (https://github.com/tuladhar/gocat)
 # 
-PROGRAM_NAME:=gocat
+PROGRAM_NAME=gocat
+INSTALL_PATH=/usr/local/bin
 
-build:
```
Comment (tuladhar) [inline Makefile:6]: Let's have `install` target separately rather than mixing both build and install.

File: Makefile [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
+install:
 	go build -o $(PROGRAM_NAME) src/main.go
+	@sudo install $(PROGRAM_NAME) $(INSTALL_PATH)/$(PROGRAM_NAME) 
+	echo "$(PROGRAM_NAME): installed at $(INSTALL_PATH)" 
 
-cleanup:
```
Comment (tuladhar) [inline Makefile:9]: Similarly, let's have `uninstall` target separately rather than mixing both cleanup and uninstall.

**Comment by tuladhar** [comment]
Thanks @vimarsh6739. I have added some comments.

**Comment by vimarsh6739** [comment]
I think all targets are ok now @tuladhar

File: Makefile [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
-PROGRAM_NAME:=gocat
+PROGRAM_NAME=gocat
+INSTALL_PATH=/usr/local/bin
 
 build:
 	go build -o $(PROGRAM_NAME) src/main.go
 
-cleanup:
+install:
+	go build -o $(PROGRAM_NAME) src/main.go
```
Comment (tuladhar) [inline Makefile]: We can re-use the build target and remove this, like so:
```
install: build
```
Now, `build` will auto run before install.

File: Makefile [hunk 5]
```diff
@@ -1,10 +1,22 @@
...
+	@sudo install $(PROGRAM_NAME) $(INSTALL_PATH)/$(PROGRAM_NAME) 
+	echo "$(PROGRAM_NAME): installed at $(INSTALL_PATH)" 
+
+clean:
+	rm $(PROGRAM_NAME)
+
+uninstall:
 	rm $(PROGRAM_NAME)
```
Comment (tuladhar) [inline Makefile]: Let's amend this similar to the above comment.

**Comment by tuladhar** [comment]
Thanks @vimarsh6739. I have added some further comments, please check. Besides that, looks good to merge! 👍

**Pull request merged**